### PR TITLE
feat: implement core log parsing infrastructure

### PR DESCRIPTION
**Pull request by myakove**
- Add LogEntry dataclass with structured log data
- Implement LogParser with regex-based GitHub webhook context extraction
- Add LogFilter with comprehensive filtering capabilities
- Support async log file monitoring and real-time streaming
- Include full test coverage with 24 passing tests

## Summary by CodeRabbit

* **New Features**
  * Introduced a comprehensive Log Viewer web interface and API for real-time and historical monitoring of webhook processing logs, including advanced filtering, PR flow visualization, dark/light theme support, and JSON export.
  * Added real-time WebSocket log streaming and interactive workflow timeline visualization.

* **Documentation**
  * Added detailed Log Viewer documentation, including usage instructions, API details, security considerations, and troubleshooting guidance.
  * Updated README with container image URL changes and formatting improvements.

* **Bug Fixes**
  * Updated rate limit handling to reflect changes in the GitHub API structure.

* **Refactor**
  * Enhanced logging throughout workflow and handler modules for improved step-level visibility and traceability.
  * Refactored pull request processing into clearer setup and CI/CD execution phases with distinct logging.
  * Centralized and standardized log prefix preparation with consistent repository color coding.
  * Removed internal repository color logic from GitHub API module, delegating to shared helper function.

* **Tests**
  * Added extensive tests for Log Viewer API, WebSocket streaming, log parsing, filtering, workflow step extraction, edge cases, memory optimization, and performance benchmarks.
  * Updated existing tests to reflect changes in rate limit attribute naming and repository color retrieval logic.
  * Added fixtures and environment optimizations to improve test reliability and performance.
  * Added tests for static files path validation and frontend performance optimizations.

* **Chores**
  * Updated dependencies and configuration files to support new features and maintain compatibility.
  * Updated .gitignore to exclude additional AI-related directories.

### CODE REVIEW

File: webhook_server/app.py [hunk 1]
```diff
@@ -276,3 +280,144 @@ async def process_with_error_handling(_api: GithubWebhook, _logger: logging.Logg
...
+    controller = LogViewerController(logger=LOGGER)
+    await controller.handle_websocket(
+        websocket=websocket,
+        hook_id=hook_id,
+        pr_number=pr_number,
+        repository=repository,
+        event_type=event_type,
+        github_user=github_user,
+        level=level,
+    )
```
[the remark this answers is not in the document]
Comment (myakove) [reply webhook_server/app.py]: Please open issue for this
